packages/auth/src/mfa/assertions/totp.test.ts

```diff
@@ -208,6 +219,91 @@ describe('core/mfa/totp/assertions/TotpMultiFactorAssertionImpl', () => {
   });
 });
 
+describe('Testing signIn Flow', () => {
+  let auth: TestAuth;
+  let assertion: MultiFactorAssertionImpl;
+  let totpSignInResponse: StartTotpMfaSignInResponse;
+  let session: MultiFactorSessionImpl;
+  beforeEach(async () => {
+    mockFetch.setUp();
+    auth = await testAuth();
+    session = MultiFactorSessionImpl._fromMfaPendingCredential(
+      'mfa-pending-credential'
+    );
+  });
+  afterEach(mockFetch.tearDown);
+
+  it('should finalize mfa signin for totp', async () => {
+    totpSignInResponse = { 'verificationCode': '123456' } as any;
+    assertion = TotpMultiFactorGenerator.assertionForSignIn(
+      'enrollment-id',
+      '123456'
+    ) as any;
+
+    const mock = mockEndpoint(
+      Endpoint.FINALIZE_MFA_SIGN_IN,
+      totpSignInResponse
+    );
+
+    const response = await assertion._process(auth, session);
+
+    expect(response).to.eql(totpSignInResponse);
+
+    expect(mock.calls[0].request).to.eql({
+      mfaPendingCredential: 'mfa-pending-credential',
+      mfaEnrollmentId: 'enrollment-id',
+      totpVerificationInfo: {
+        verificationCode: '123456'
+      }
+    });
+  });
+
+  it('should throw Firebase Error if enrollment-id is undefined', async () => {
+    let enrollmentId: string;
+    let response: any;
+    totpSignInResponse = { 'verificationCode': '123456' } as any;
+    assertion = TotpMultiFactorGenerator.assertionForSignIn(
+      enrollmentId,
+      '123456'
+    ) as any;
+
+    const mock = mockEndpoint(
+      Endpoint.FINALIZE_MFA_SIGN_IN,
+      totpSignInResponse
+    );
+    //console.log("response: ",await assertion._process(auth,session));
+
+    try {
```

Comment:
For some reason, I was not able to do this with the chaiAsPromised format of using `await expect(foo()).to.eventually.be.eql(something)` and so resolved to default try, catch

Reply:
This is expected.. you need to wrap await inside try/catch to catch errors.

Reply:
I'd recommend using the `.to.be.rejectedWith()` construction we use elsewhere in the SDK. It's a bit cleaner than using try/catch, plus as the test is currently written if the assertion._process() _doesn't_ error as expected, the assertions in the catch block will never run. Here's an example:

https://github.com/firebase/firebase-js-sdk/blob/48a127b2f42efac369735f83e2597db08b24a0f9/packages/auth/src/api/index.test.ts#L161-L170

So in your case here, it would be:
```ts
await expect(assertion._process(auth, session)).to.be.rejectedWith('auth/argument-error');
```

If that's not working we should try to figure out why it's not

Reply:
I have tried this again and it does not work.

Reply:
The difference between the given example and this test is: just throwing error and throwing an error in an aync function. The latter is same as rejecting a promise. I noticed that if I use
```return new Promise(()=>{
     throw new Error('test')
     })
 ```
 
 ```await expect(assertion._process(auth, session)).to.be.rejectedWith('test');``` works

Reply:
See my comment at https://github.com/firebase/firebase-js-sdk/pull/6626/files#r993768335 and try the promise-based method again, that should fix the issue